File: test-suite-kotlin/src/test/kotlin/io/micronaut/kafka/docs/producer/config/RandomTransactionalIdClient.kt
package io.micronaut.kafka.docs.producer.config

import io.micronaut.configuration.kafka.annotation.KafkaClient
import io.micronaut.context.annotation.Prototype
import io.micronaut.context.annotation.Requires

@Requires(property = "spec.name", value = "RandomTransactionalIdClientTest")
// tag::clazz[]
@Prototype
@KafkaClient(id = "my-client", transactionalId = "my-tx-id-\${random.uuid}")
interface RandomTransactionalIdClient {
    // define client API
}
// end::clazz[]

File: test-suite-python/src/test/python/micronaut/kafka/docs/producer/config/RandomTransactionalIdClient.py
from abc import ABC

from micronaut.configuration.kafka.annotation import KafkaClient
from micronaut.context.annotation import Prototype, Requires


@Requires(property="spec.name", value="RandomTransactionalIdClientTest")
# tag::clazz[]
@Prototype
@KafkaClient(id="my-client", transactionalId="my-tx-id-${random.uuid}")
class RandomTransactionalIdClient(ABC):
    # define client API
    pass
# end::clazz[]
